dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaAPIHostsTask.java

```diff
@@ -63,14 +64,21 @@ protected void doRun(
         JsonNode jsonHosts;
         try (CloseableHttpResponse hostsResponse =
             httpClient.execute(new HttpGet(hostPerClusterUrl))) {
-          jsonHosts = getObjectMapper().readTree(hostsResponse.getEntity().getContent());
+          final int statusCode = hostsResponse.getStatusLine().getStatusCode();
+          if (!isStatusCodeOK(statusCode)) {
+            throw new MetadataDumperUsageException(
+                String.format(
+                    "Cloudera Error: Response status code is %d but 2xx is expected.", statusCode));
+          }
+          jsonHosts = readJsonTree(hostsResponse.getEntity().getContent());
+        } catch (JsonParseException error) {
+          throw new MetadataDumperUsageException("Cloudera Error:" + error.getMessage());
```

Comment:
I'm curious if we lose some important information here. 
It looks like we need an additional constructor `public MetadataDumperUsageException(String message, Throwable cause)`

Reply:
Why can't we pass all important information in context of the exception message? If we pass `Throwable cause` then it will be necessary to check all possible exception types when `MetadataDumperUsageException` will be handled.

Reply:
It is possible, but you will lose the stack trace here at least. Sometimes losing the stack trace is acceptable.

If you create a new constructor, you won't need to change anything.

Reply:
Okay, but I think we don't need to add this parameter into the constructor. There is a specific API in Throwable class: https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#initCause-java.lang.Throwable-